### PR TITLE
feat: LT-1019 Edit Generated Document

### PR DESCRIPTION
**Pull request by BenAndrew-Kainos**

### CODE REVIEW

File: cases/generated_documents/helpers.py [hunk 1]
```diff
@@ -27,14 +28,23 @@ def get_letter_template_for_case(id, case):
...
-def get_letter_template(id):
-    """
-    Returns a letter template or returns a 404 on failure
-    """
-    try:
-        return LetterTemplate.objects.get(pk=id)
-    except LetterTemplate.DoesNotExist:
-        raise NotFoundError({"letter_template": LetterTemplatesPage.NOT_FOUND_ERROR})
+def get_generated_document_data(request_params, pk):
+    tpk = request_params.get("template")
```
Comment (a-bek) [inline cases/generated_documents/helpers.py]: Could you rename to `template_id`?

File: cases/generated_documents/helpers.py [hunk 2]
```diff
@@ -27,14 +28,23 @@ def get_letter_template_for_case(id, case):
     try:
         return LetterTemplate.objects.get(pk=id, case_types__id=case.type)
     except LetterTemplate.DoesNotExist:
-        raise NotFoundError({"letter_template": LetterTemplatesPage.NOT_FOUND_ERROR})
+        raise NotFoundError({"letter_template": GeneratedDocumentsEndpoint.LETTER_TEMPLATE_NOT_FOUND})
```
Comment (a-bek) [inline cases/generated_documents/helpers.py]: Could you change the function parameter `(id, case)` to `(template_id, case)`?

File: cases/generated_documents/helpers.py [hunk 3]
```diff
@@ -27,14 +28,23 @@ def get_letter_template_for_case(id, case):
...
 
-def get_letter_template(id):
-    """
-    Returns a letter template or returns a 404 on failure
-    """
-    try:
-        return LetterTemplate.objects.get(pk=id)
-    except LetterTemplate.DoesNotExist:
-        raise NotFoundError({"letter_template": LetterTemplatesPage.NOT_FOUND_ERROR})
+def get_generated_document_data(request_params, pk):
```
Comment (a-bek) [inline cases/generated_documents/helpers.py]: Could we create some structure in the return from this function?

ie:
```
# Define the schema somewhere for documentdata.
DocumentData = namedtuple('data', 'error case template document_html text')

def get_generated_document_data(...):
    return DocumentData(error=None, case=case, template=template, html=document_html, text=text)
```

Comment (BenAndrew-Kainos) [reply cases/generated_documents/helpers.py]: Yeah I like that

File: cases/generated_documents/helpers.py (hunk 2 above)
Comment (BenAndrew-Kainos) [reply cases/generated_documents/helpers.py]: Done

File: cases/generated_documents/helpers.py (hunk 1 above)
Comment (BenAndrew-Kainos) [reply cases/generated_documents/helpers.py]: Done

File: cases/generated_documents/helpers.py (hunk 3 above)
Comment (BenAndrew-Kainos) [reply cases/generated_documents/helpers.py]: Done

File: letter_templates/tests/tests_view.py [hunk 4]
```diff
@@ -67,3 +67,26 @@ def test_get_letter_template_success(self):
         self.assertIn(CaseTypeEnum.END_USER_ADVISORY_QUERY, str(template["case_types"]))
         self.assertIsNotNone(template.get("created_at"))
         self.assertIsNotNone(template.get("last_modified_at"))
+
+    def test_get_letter_template_with_preview_success(self):
+        url = reverse("letter_templates:letter_template", kwargs={"pk": str(self.letter_template.id)})
+        url += "?generate_preview=True"
+        response = self.client.get(url, **self.gov_headers)
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply letter_templates/tests/tests_view.py]: sure

File: letter_templates/views.py [hunk 5]
```diff
@@ -48,12 +48,19 @@ def get(self, request, *args, **kwargs):
...
+        )
+        paragraph_text = get_paragraphs_as_html(paragraphs)
 
         if "generate_preview" in request.GET and bool(request.GET["generate_preview"]):
-            data["preview"] = get_preview(template=template_object)
+            data["preview"] = generate_preview(layout=template_object.layout.filename, text=paragraph_text)
             if "error" in data["preview"]:
                 return JsonResponse(data=data["preview"], status=status.HTTP_400_BAD_REQUEST)
 
+        if "text" in request.GET and bool(request.GET["text"]):
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply letter_templates/views.py]: sure

File: letter_templates/views.py [hunk 6]
```diff
@@ -48,12 +49,19 @@ def get(self, request, *args, **kwargs):
...
+        paragraph_text = get_paragraphs_as_html(paragraphs)
 
-        if "generate_preview" in request.GET and bool(request.GET["generate_preview"]):
-            data["preview"] = get_preview(template=template_object)
+        if "generate_preview" in request.GET and str_to_bool(request.GET["generate_preview"]):
+            data["preview"] = generate_preview(layout=template_object.layout.filename, text=paragraph_text)
             if "error" in data["preview"]:
                 return JsonResponse(data=data["preview"], status=status.HTTP_400_BAD_REQUEST)
 
+        if "text" in request.GET and str_to_bool(request.GET["text"]):
```
Comment (andytaylorkainos) [inline letter_templates/views.py]: if you use `str_to_bool(request.GET.get("text"))` and drop `"text" in request.GET` from the condition. all your scenario's should be handled

Comment (BenAndrew-Kainos) [reply letter_templates/views.py]: Ok I trust you

Comment (BenAndrew-Kainos) [reply letter_templates/views.py]: Done

**Pull request merged**